### PR TITLE
fix: update docs and add code examples

### LINKED ISSUE #150: [graph-sdk] Add docs/examples for Rust & Java
Providers & other developers will require example code to show how to interact with GraphSDK.

### PR DESCRIPTION
**Pull request by JoeCap08055**
# Goal
The goal of this PR is to add a `docs/` directory with diagrams and code examples.

Closes #150

### CODE REVIEW

File: docs/examples/java-examples.md [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+        // substring use is to skip the `0x` in front of the payload
+        .setContent(ByteString.fromHex(page.payload.substring(2)))
+        .build()
+    }
+    val bundles = ImportBundles.newBuilder()
+      .addBundles(
+        ImportBundles.ImportBundle.newBuilder()
+          .setDsnpUserId(msaId.toLong())
+          .setSchemaId(publicFollowSchemaId)
+          .setDsnpKeys(DsnpKeys.newBuilder().setDsnpUserId(msaId.toLong()).build())
```
Comment (wilwade) [inline docs/examples/java-examples.md:132]: I don't think we need this line for public graph do we?
```suggestion
```

File: docs/examples/java-examples.md [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+
+    // ----- PUBLISH TO BLOCKCHAIN ----- //
+    for (update in updates) {
+      if (update.hasPersist()) {
+        val persist = update.persist
+        // finally, update the graph on chain with new connections
+        val retVal =
+          frequencyClient.createUpsertPage(
+            msaId,
+            publicFollowSchemaId,
```
Comment (wilwade) [inline docs/examples/java-examples.md:169]: My understanding is that all the values come from persist.

```suggestion
            persist.ownerDsnpUserId,
            persist.schemaId,
```

File: docs/examples/java-examples.md [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+
+    // ----- SETUP GRAPH ----- //
+    // initialize the Graph in memory
+    Configuration configuration = Configuration.getMainnet()
+    val graph = Graph(configuration)
+
+    // Get the schema ID for the graph keys
+    val schemaId = configuration
+    .getEnvironment()
+    .getConfig()
```
Comment (wilwade) [inline docs/examples/java-examples.md:35]: Is this right? Feels off to be calling .getConfig on a variable of type Configuration

File: docs/examples/java-examples.md [hunk 4]
```diff
@@ -0,0 +1,393 @@
...
+
+    // ----- PUBLISH TO BLOCKCHAIN ----- //
+    for (update in updates) {
+      if (update.hasPersist()) {
+        val persist = update.persist
+        // finally, updating the graph on chain with new connections
+        val retVal =
+          frequencyClient.createUpsertPage(
+            msaId,
+            privateFollowSchemaId,
```
Comment (wilwade) [inline docs/examples/java-examples.md:264]: persist.[value]

File: docs/examples/java-examples.md [hunk 5]
```diff
@@ -0,0 +1,393 @@
...
+
+    // ----- PUBLISH TO BLOCKCHAIN ----- //
+    for (update in updates) {
+      if (update.hasPersist()) {
+        val persist = update.persist
+        // finally, updating the graph on chain with new connections
+        val retVal =
+          frequencyClient.createUpsertPage(
+            msaId,
+            privateFriendshipSchemaId,
```
Comment (wilwade) [inline docs/examples/java-examples.md:378]: persist.[value]

**Review by wilwade — CHANGES_REQUESTED** [review]

File: docs/examples/java-examples.md [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+      AddItemAction.from(update.addKey.payload.toByteArray())
+    }
+
+    // finally, add the new exported key to blockchain
+    // here we are using `createApplyItemActions` for simplicity but in reality this should be done using
+    // `createApplyItemActionsWithSignature` since user should sign the payload of adding a new graph key with one of
+    // their msa control keys
+    val retVal =
+      frequencyClient.createApplyItemActions(graphOwnerMsaId, publicKeySchemaId, keyContentHash.toBigInteger(), itemizedActions)
+        .join()
```
Comment (wilwade) [inline docs/examples/java-examples.md:142]:
```suggestion
    // finally, add the new exported key to blockchain
    // the user MUST sign the payload of adding a new graph key with one of their msa control keys
    // The full data to sign is available: https://libertydsnp.github.io/frequency/pallet_stateful_storage/types/struct.ItemizedSignaturePayload.html
    val retVal =
      frequencyClient.createApplyItemActionsWithSignature(graphOwnerSigningPublicKey, signatureProof, publicKeySchemaId, keyContentHash.toBigInteger(), itemizedActions)
        .join()
```

**Review by wilwade — APPROVED** [review]
Two small optional changes

File: docs/examples/java-examples.md (hunk 6 above)
Comment (JoeCap08055) [reply docs/examples/java-examples.md:142]: Ah, good catch--I forgot the GraphPublicKey schema was `SignatureRequired` in addition to `AppendOnly`

**Pull request merged**